AsyncDisplayKit/ASCollectionView.mm

```diff
@@ -357,6 +382,14 @@ - (void)performBatchUpdates:(void (^)())updates completion:(void (^)(BOOL))compl
   [self performBatchAnimated:YES updates:updates completion:completion];
 }
 
+- (void)registerSupplementaryNodeOfKind:(NSString *)elementKind
+{
+  ASDisplayNodeAssert(elementKind != nil, @"A kind is needed for supplementary node registration");
+  [_registeredSupplementaryKinds addObject:elementKind];
+  [self registerClass:[UICollectionReusableView class] forSupplementaryViewOfKind:elementKind
```

Comment:
@levi in the meantime, an AS subclass of UICollectionReusableView has been created.  Please adopt it as it would be required to ensure setSelected / setHighlighted (added recently) are not broken.